src/instruments/src/SD/Pages/Eng/Eng.tsx

```diff
@@ -0,0 +1,312 @@
+    const [n2Igniting] = useSimVar(`TURB ENG IS IGNITING:${engineNumber}`, 'bool', 300);
+    const [showIgniter, setShowIgniter] = useState(false);
+    const [engSelectorPosition] = useSimVar('L:XMLVAR_ENG_MODE_SEL', 'Enum', 1000);
+
+    const [weightUnit] = usePersistentProperty('CONFIG_USING_METRIC_UNIT', '1');
+
+    const [flightPhase] = useSimVar('L:A32NX_FWC_FLIGHT_PHASE', 'number', 4000);
+    const [shouldPressurePulse, setShouldPressurePulse] = useState(false);
+    const [shouldQuantityPulse, setShouldQuantityPulse] = useState(false);
+
+    const [fuelUsed] = useSimVar(`L:A32NX_FUEL_USED:${engineNumber}`, 'number', 500);
+    // Fuel used has a step of 10 when in Kilograms and 20 when in imperial pounds
+    const displayedFuelUsed = parseInt(weightUnit) === 1 ? Math.round(fuelUsed / 10) * 10 : Math.round(fuelUsed * 2.20462 / 20) * 20;
+
+    const [engineOilQuantity] = useSimVar(`ENG OIL QUANTITY:${engineNumber}`, 'percent', 100);
+    const OIL_QTY_MAX = 24.25;
+    const OIL_QTY_LOW_ADVISORY = 1.35;
+    const displayedEngineOilQuantity = engineOilQuantity === 100 ? OIL_QTY_MAX : Math.round((engineOilQuantity / 100) * OIL_QTY_MAX / 0.5) * 0.5; // Engine oil quantity has a step of 0.2
+    const [quantityAtOrBelowLow, setQuantityAtOrBelowLow] = useState(false);
+
+    const [engineOilPressure] = useSimVar(`ENG OIL PRESSURE:${engineNumber}`, 'psi', 200);
+    const displayedEngineOilPressure = Math.round(engineOilPressure / 2) * 2; // Engine oil pressure has a step of 2
+    const OIL_PSI_MAX = 130;
+    const OIL_PSI_HIGH_LIMIT = 130;
+    const OIL_PSI_LOW_LIMIT = 14; // TODO FIXME: standin value
+    const OIL_PSI_VLOW_LIMIT = 12;
+    const [psiNeedleRed, setPsiNeedleRed] = useState(true);
+    const [pressureAboveHigh, setPressureAboveHigh] = useState(false);
+    const [pressureBelowLow, setPressureBelowLow] = useState(false);
+
+    const [engineOilTemperature] = useSimVar(`GENERAL ENG OIL TEMPERATURE:${engineNumber}`, 'celsius', 250);
+    const OIL_TEMP_HIGH_ADVISORY = 140;
+    const OIL_TEMP_VHIGH_LIMIT = 155;
+    const displayedEngineOilTemperature = Math.round(engineOilTemperature / 5) * 5; // Engine oil temperature has a step of 5
+    const [tempAmber, setTempAmber] = useState(false);
+    const [shouldTemperaturePulse, setShouldTemperaturePulse] = useState(false);
+    const [tempBeenAboveAdvisory, setTempBeenAboveAdvisory] = useState(false);
+
+    const [n1Vibration] = useSimVar(`TURB ENG VIBRATION:${engineNumber}`, 'Number');
+
+    const [n2Vibration] = useSimVar(`TURB ENG VIBRATION:${engineNumber}`, 'Number'); // FIXME TODO: should have a different value than N1, currently API limited
+
+    function getNeedleValue(value: any, max: number): number {
+        const numberValue = (Number)(value);
+        if (numberValue < max) {
+            return (numberValue / max) * 100;
+        }
+        return 100;
+    }
+
+    useEffect(() => {
```

Comment:
**issue:** This `useEffect` is long. It should have a comment above describing what effects it is handling.